Fix template docs for v4 (#665)
strapi/documentation: docs/developer-docs/latest/setup-deployment-guides/installation/templates.md

* A template should follow a specific [file structure](#file-structure), which can be created manually or automatically generated with the [`strapi templates:generate` command](/developer-docs/latest/developer-resources/cli/CLI.md#strapi-templates-generate):

  <code-group>

    <code-block title="YARN">
      yarn strapi templates:generate <path>
    </code-block>

    <code-block title="NPM">
      npx strapi templates:generate <path>
    </code-block>

  </code-group>

### File structure

The root of the template repository can include as many files as necessary but must include at least:

* a `template` folder, 
* a `package.json` file,
* and a `template.json` file.

The `template.json` is used to extend the Strapi application's default `package.json`. All properties overwriting the default `package.json` should be included in a root `package` property:

```json
// path: ./template.json

{
  "package": {
    "dependencies": {

>       "strapi-plugin-graphql": "^4.0.0"

Author: ```suggestion
      "@strapi/plugin-graphql": "^4.0.0"
```

> Templates and starters are not the same thing:

Author: Sounds good but FYI I don't see any mention of `::: strapi` on the styleguide

### File structure

> You can add as many files as you want to the root of your template repository. But it must at least have `template` directory, a `package.json`, and a `template.json` file.

Reviewer: ```suggestion
The root of the template repository can include as many files as necessary but must include at least:

* a `template` folder, 
* a `package.json` file,
* and a `template.json` file.
```
